gax/src/main/java/com/google/api/gax/batching/v2/BatcherImpl.java

```diff
@@ -115,13 +131,22 @@ public void flush() throws InterruptedException {
     awaitAllOutstandingBatches();
   }
 
-  /** Sends accumulated elements asynchronously for batching. */
+  /**
+   * Sends accumulated elements asynchronously for batching.
+   *
+   * <p>Note: This method can be invoked concurrently unlike {@link #add} and {@link #close}, which
+   * can only be called from single user thread. Please take caution to avoid race condition.
+   */
   private void sendBatch() {
-    if (currentOpenBatch == null) {
-      return;
+
+    final Batch<ElementT, ElementResultT, RequestT, ResponseT> accumulatedBatch;
+    synchronized (elementLock) {
+      if (currentOpenBatch.isEmpty()) {
+        return;
+      }
+      accumulatedBatch = currentOpenBatch;
+      currentOpenBatch = new Batch<>(prototype, batchingDescriptor, batchingSettings);
```

Comment:
what is `currentOpenBatch` used for?

Reply:
`currentOpenBatch` is an instance variable, it's used by add() to accumulate elements.